raft/src/main/java/org/apache/kafka/raft/ReplicatedCounter.java

```diff
@@ -68,20 +70,65 @@ public synchronized void increment() {
     @Override
     public synchronized void handleCommit(BatchReader<Integer> reader) {
         try {
-            int initialValue = this.committed;
+            int initialValue = committed;
             while (reader.hasNext()) {
                 BatchReader.Batch<Integer> batch = reader.next();
                 log.debug("Handle commit of batch with records {} at base offset {}",
                     batch.records(), batch.baseOffset());
                 for (Integer value : batch.records()) {
-                    if (value != this.committed + 1) {
-                        throw new AssertionError("Expected next committed value to be " +
-                            (this.committed + 1) + ", but instead found " + value + " on node " + nodeId);
+                    if (value != committed + 1) {
```

Comment:
nit: not really part of this PR, but can we use variable names besides "value" and "initialValue"?

Reply:
Done. Not sure if it is any better :smile: